rest/src/main/scala/io/udash/rest/openapi/RestStructure.scala

```diff
@@ -151,8 +151,10 @@ object RestStructure extends AdtMetadataCompanion[RestStructure] {
   ) extends TypedMetadata[T] {
 
     val fallbackValue: Opt[JsonValue] =
-      (whenAbsentInfo.map(_.fallbackValue) orElse defaultValueInfo.map(_.fallbackValue)).flatten
-    val hasFallbackValue: Boolean = fallbackValue.isDefined
+      if (info.optional) Opt.Empty
+      else (whenAbsentInfo.map(_.fallbackValue) orElse defaultValueInfo.map(_.fallbackValue)).flatten
```

Comment:
The tests need some love. An obvious (and intentionally not used by you here?) refactoring to `whenAbsentInfo.flatMap(_.fallbackValue) orElse defaultValueInfo.flatMap(_.fallbackValue)` passes.

Reply:
Ok, it seems that other mechanisms make it so that this check is not necessary but I'd leave it anyway to express the intent better and make it safe against possible changes in other parts of the macro system.